source/neuropod/bindings/java/src/main/java/com/uber/neuropod/LibraryLoader.java

```diff
@@ -66,32 +96,101 @@ private static boolean loadSharedLibrary() {
+                if (embeddedLibFile != null) {
+                    LOGGER.log(Level.INFO, "Extracted embedded lib file {0}", embeddedLibFile);
+                }
             }
+
             for (String binName : BIN_NAMES) {
                 File binFile = extractFile(libAbsPath, resPath, binName);
                 if (binFile != null) {
+                    LOGGER.log(Level.INFO, "Extracted bin file {0}", binFile);
                     binFile.setExecutable(true);
                 }
             }
+
+            // Resources may have backend packages (optional) that should e extracted and installed then.
+            // If NEUROPOD_BASE_DIR is set, use this path to install backends.
+            // Note that packages are tar.gz files that has "neuropod standard" directory structure, for instance:
+            // 0.2.0/backends/tensorflow_1.15.0/
+            // 0.2.0/backends/torchscript_1.4.0/
+            // "register backends" looks at NEUROPOD_BASE_DIR and register backends by type and version.
+            // Note: NEUROPOD_BASE_DIR may already have other backends that will be registered also.
+            // When Neuropod model is loaded, backend that has best match is used. It means that embedded backends
+            // that are installed here are not necessaraly best match and may not used at all.
+            String neuropodBaseDir = System.getenv(NEUROPOD_BASE_DIR_ENV_VAR);
+            if (neuropodBaseDir == null) {
+                neuropodBaseDir = NEUROPOD_BASE_DIR_DEFAULT;
+            }
+            for (String libName : EMBEDDED_BACKEND_PACKAGES_NAMES) {
+                File embeddedPackageFile = extractFile(libAbsPath, resPath, libName);
+                if (embeddedPackageFile != null) {
+                    LOGGER.log(Level.INFO, "Extracted embedded package file {0}", embeddedPackageFile);
+                    String fileName = embeddedPackageFile.getName();
+                    if (fileName.equals(TENSORFLOW_BACKEND) || fileName.equals(TORCHSCRIPT_BACKEND)) {
+                        installFramework(libAbsPath, fileName, neuropodBaseDir);
+                    }
+                }
+            }
+
             File libFile = extractFile(libAbsPath, resPath, System.mapLibraryName(JNI_NAME));
-            System.load(libFile.getCanonicalPath());
+            if (libFile != null) {
+                System.load(libFile.getCanonicalPath());
+            }
+
             nativeExport(libAbsPath);
+            result = true;
         } catch (IOException e) {
-            System.out.println(e.getMessage());
+            LOGGER.log(Level.WARNING, e.getMessage());
+            result = false;
+        }
+        return result;
+    }
+
+    private static boolean installFramework(String libAbsPath, String frameworkPackage, String neuropodBaseDir) {
+        return startShellCommand(String.format("tar -xf %s/%s -C %s", libAbsPath, frameworkPackage, neuropodBaseDir));
```

Comment:
this is going to take some time + CPU usage at runtime. is this okay?

Reply:
Yes, I measured and extract+load  may take up to 10s on my mac. Good that this happens once only when we initialize API (so at start of service)

Reply:
Decided to remove "backend" part from this PR for now and pack "core" only.